### PR TITLE
No vault addr dont panic

### PR DESCRIPTION
**Pull request by drmdrew**
A second attempt at a fix for #84. PTAL @hairyhenderson 

This time, rather than twisting gomplate into a pretzel trying to propagate errors, a small testing hack is being made to allow `log.Fatal` to be overwritten inside tests to inject a mock/spy that doesn't cause the process to exit.

Added unit tests and also checked on the command-line that gomplate doesn't panic when some/all of the `VAULT_*` environment variables are missing:

```
echo -n '{{(datasource "vault" "secret/foo").value}}' | gomplate -d vault="vault://"
2016/11/22 09:49:45 VAULT_ADDR is an unparseable URL!

export VAULT_ADDR=https://localhost:8500
echo -n '{{(datasource "vault" "secret/foo").value}}' | gomplate -d vault="vault://"
2016/11/22 09:49:37 No vault auth strategy configured
```

### CODE REVIEW

File: vault/log.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package vault
```
Comment (hairyhenderson) [inline vault/log.go]: can you throw these into `client_test.go`?

**Comment by drmdrew** [comment]
Removed `log.go`

File: vault/client.go [hunk 2]
```diff
@@ -81,9 +89,13 @@ func (c *Client) Do(req *http.Request) (*http.Response, error) {
 
 // Login - log in to Vault with the discovered auth backend and save the token
 func (c *Client) Login() error {
+	if c.Auth == nil {
```
Comment (hairyhenderson) [inline vault/client.go]: Can you move this check inside `getAuthStrategy()`?

**Comment by drmdrew** [comment]
Moved `logFatal` to `getAuthStrategy`

**Pull request merged**